### DISCUSSION TITLE
Is min-batch applicable to all graph neural network models?

### DISCUSSION THREAD

**Discussion by forDuan** [pyg-team/pytorch_geometric#9264] [Q&A] [answered]
I am using a graph neural network model for the task of graph classification, and I have packaged the graphs in the dataset in batches into torch_geometric.data.batch.Batch, and then input it into GCNConv for graph classification. 
Now I have learned about many other GNN models, such as GAT, GATv2, TransFormerConv, and so on. But I have not fully understood the principles of these models yet.
so I would like to know if I can also package the graphs in the dataset as Batch and input it into these models for graph classification? Is the data structure output by the model the same as GCN?

**Comment by rusty1s** [comment, marked as answer]
Yes, this is very similar. All GNN layers are just computing node-level representations, and they should be easy to replace with other GNN layers. The final readout stage/classifier stays the same in all these scenarios.

**Reply by forDuan** [reply]
Thank you!
